pytensor/tensor/rewriting/elemwise.py

```diff
@@ -529,333 +539,486 @@ def local_upcast_elemwise_constant_inputs(fgraph, node):
+                        # No client was actually fuseable
+                        for client in fuseable_clients.pop(starting_out, []):
+                            unfuseable_clients[starting_out].append(client)
+
+                    else:
+                        yield subgraph_inputs, subgraph_outputs
+
+                        # This is where we avoid repeated work by using a stateful
+                        # generator. For large models (as in `TestFusion.test_big_fusion`)
+                        # this can provide huge speedups
+
+                        # Update fuseable mappings
+                        next_nodes = fg.apply_nodes
+                        (new_composite_node,) = next_nodes - starting_nodes
+                        dropped_nodes = starting_nodes - next_nodes
+
+                        # Remove intermediate Composite nodes from mappings
+                        for dropped_node in dropped_nodes:
+                            (dropped_out,) = dropped_node.outputs
+                            fuseable_clients.pop(dropped_out, None)
+                            unfuseable_clients.pop(dropped_out, None)
+                            visited_nodes.remove(dropped_node)
+
+                        # Any input is now definitely unfuseable
+                        for inp in subgraph_inputs:
+                            if inp in fuseable_clients:
+                                new_fuseable_clients = [
+                                    client
+                                    for client in fuseable_clients[inp]
+                                    if client not in dropped_nodes
+                                ]
+                                if new_fuseable_clients:
+                                    fuseable_clients[inp] = new_fuseable_clients
+                                else:
+                                    fuseable_clients.pop(inp)
+                            unfuseable_clients[inp] = [
+                                client
+                                for client in unfuseable_clients[inp]
+                                if client not in dropped_nodes
+                            ] + [new_composite_node]
+
+                        # Any client is now definitely unfuseable
+                        for out in new_composite_node.outputs:
+                            unfuseable_clients[out] = [
+                                client for client, _ in fg.clients[out]
+                            ]
+                        visited_nodes.add(new_composite_node)
+                    break
+                else:  # nobreak
+                    return
 
-    """
+        # pytensor.dprint(fgraph, print_type=True)
+        for res in find_next_fuseable_subgraph(fgraph):
+            # print(f">> >> Start of iteration {nb_replacement}: {len(fgraph.apply_nodes)=}")
+            if res is None:
+                # print("<< No further fuseable subgraph found")
```

Comment:
With all these debugging prints, should this rewriter maybe get a `verbosity` setting and log to a real logger if asked to?

Reply:
I don't think they would be useful outside of the context of actually working on the rewrite. I removed them